src/core/core.scale.js

```diff
@@ -93,10 +83,125 @@ function getPixelForGridLine(scale, index, offsetGridLines) {
 	return lineValue;
 }
 
-function computeTextSize(context, tick, font) {
-	return helpers.isArray(tick) ?
-		helpers.longestText(context, font, tick) :
-		context.measureText(tick).width;
+/**
+ * Returns {width, height, offset} objects for the first, last, widest, highest tick
+ * labels where offset indicates the anchor point offset from the top in pixels.
+ */
+function computeLabelSizes(ctx, tickFonts, ticks, cache) {
+	var length = ticks.length;
+	var widths = [];
+	var heights = [];
+	var offsets = [];
+	var i, j, jlen, label, tickFont, fontString, data, gc, lineHeight, width, height, nestedLabel, widest, highest;
+
+	helpers.each(tickFonts, function(font) {
+		cache[font.string] = cache[font.string] || {data: {}, garbageCollect: []};
+	});
+
+	for (i = 0; i < length; ++i) {
+		label = ticks[i].label;
+		tickFont = ticks[i].major ? tickFonts.major : tickFonts.minor;
+		ctx.font = fontString = tickFont.string;
+		data = cache[fontString].data;
+		gc = cache[fontString].garbageCollect;
+		lineHeight = tickFont.lineHeight;
+		width = height = 0;
+		// Undefined labels and arrays should not be measured
+		if (!helpers.isNullOrUndef(label) && !helpers.isArray(label)) {
+			width = helpers.measureText(ctx, data, gc, width, label);
+			height = lineHeight;
+		} else if (helpers.isArray(label)) {
+			// if it is an array lets measure each element
+			for (j = 0, jlen = label.length; j < jlen; ++j) {
+				nestedLabel = label[j];
+				// Undefined labels and arrays should not be measured
+				if (!helpers.isNullOrUndef(nestedLabel) && !helpers.isArray(nestedLabel)) {
+					width = helpers.measureText(ctx, data, gc, width, nestedLabel);
+					height += lineHeight;
+				}
+			}
+		}
+		widths.push(width);
+		heights.push(height);
+		offsets.push(lineHeight / 2);
+	}
+
+	helpers.each(cache, function(fontCache) {
+		var garbageCollect = fontCache.garbageCollect;
+		var gcLen = garbageCollect.length / 2;
+		if (gcLen > length) {
+			for (i = 0; i < gcLen; ++i) {
+				delete fontCache.data[garbageCollect[i]];
+			}
+			garbageCollect.splice(0, gcLen);
+		}
+	});
+
+	widest = widths.indexOf(Math.max.apply(null, widths));
+	highest = heights.indexOf(Math.max.apply(null, heights));
+
+	return {
```

Comment:
could add a internal helper here for object creation (brevity(?) vs performance):
```
function valueAt(idx) {
	return {
		width: widths[idx] || 0,
		height: heights[idx] || 0,
		offset: offsets[idx] || 0
	};
}
return {
	first: valueAt(0),
	last: valueAt(length - 1),
	widest: valueAt(widest),
	highest: valueAt(highest)
};
```
don't just do it, lets wait what others say about it.

Reply:
I think it's a good idea

Reply:
Done